packages/arb-token-bridge-ui/src/components/TransactionsTable/TransactionsTable.tsx

```diff
@@ -30,87 +45,244 @@ function EmptyTableRow({ children }: { children: React.ReactNode }) {
+        alt="Moon walking Arbibaut"
+        className="lg:max-h-[50%] lg:max-w-[50%]"
+      />
+    </div>
+  )
+}
+
 export type TransactionsDataStatus = 'loading' | 'error' | 'success'
 
 export type TransactionsTableProps = {
-  status: TransactionsDataStatus
-  transactions: MergedTransaction[]
-  className?: string
+  type: 'deposits' | 'withdrawals'
 }
 
-export function TransactionsTable({
-  status,
-  transactions,
-  className = ''
-}: TransactionsTableProps) {
-  const { isSmartContractWallet } = useNetworksAndSigners()
+function validate_txhash(addr: string) {
+  return /^0x([A-Fa-f0-9]{64})$/.test(addr)
+}
+
+export function TransactionsTable({ type }: TransactionsTableProps) {
+  const {
+    app: {
+      arbTokenBridge: { walletAddress }
+    }
+  } = useAppState()
+
+  const { l1, l2, isSmartContractWallet } = useNetworksAndSigners()
+
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<boolean>(false)
+  const [transactions, setTransactions] = useState<MergedTransaction[]>([])
+  const [pageParams, setPageParams] = useState<PageParams>({
+    searchString: '',
+    pageNumber: 0,
+    pageSize: 10
+  })
+  const gatewaysToUse = useGateways()
+  const [searchString, setSearchString] = useState(pageParams.searchString)
+  const [searchError, setSearchError] = useState(false)
+
+  const search = () => {
+    if (searchString && !validate_txhash(searchString)) {
+      setSearchError(true)
+      return
+    }
+    // search logic - using `searchString`
+    setPageParams(prevParams => ({
+      ...prevParams,
+      pageNumber: 0,
+      pageSize: 10,
+      searchString
+    }))
+  }
+
+  const next = () => {
+    setPageParams(prevParams => ({
+      ...prevParams,
+      pageNumber: (prevParams?.pageNumber || 0) + 1
+    }))
+  }
+
+  const prev = () => {
+    setPageParams(prevParams => ({
+      ...prevParams,
+      pageNumber: (prevParams?.pageNumber || 1) - 1
+    }))
+  }
+
+  const fetchTransactions = useCallback(async () => {
+    setLoading(true)
+    setError(false)
+    try {
+      if (type === 'deposits') {
+        const transactions = await fetchDeposits({
+          address: walletAddress,
+          l1Provider: l1.provider,
+          l2Provider: l2.provider,
+          ...pageParams
+        })
+
+        setTransactions(transformDeposits(transactions))
+      } else {
+        const transactions = await fetchWithdrawals({
+          address: walletAddress,
+          l1Provider: l1.provider,
+          l2Provider: l2.provider,
+          ...pageParams,
+          gatewayAddresses: gatewaysToUse
+        })
+        setTransactions(transformWithdrawals(transactions))
+      }
+      setLoading(false)
+    } catch (e) {
+      setLoading(false)
+      setError(true)
+    }
+  }, [pageParams, type])
+
+  useEffect(() => {
+    fetchTransactions()
+  }, [pageParams, l1, l2])
+
+  const status = loading ? 'loading' : error ? 'error' : 'success'
```

Comment:
looks like statuses are used in multiple places, should we move them to enums?

Reply:
Done